Sources/SwiftDocC/Infrastructure/DocumentationContext.swift

```diff
@@ -410,12 +410,14 @@ public class DocumentationContext: DocumentationContextDataProviderDelegate {
     func preResolveModuleNames() {
         for reference in rootModules {
             if let node = try? entity(with: reference) {
+                let displayName: String
                 switch node.name {
                 case .conceptual(let title):
-                    moduleNameCache[reference] = title
+                    displayName = title
                 case .symbol(let declaration):
-                    moduleNameCache[reference] = declaration.tokens.map { $0.description }.joined()
+                    displayName = declaration.tokens.map { $0.description }.joined()
                 }
+                moduleNameCache[reference] = (displayName, node.symbol?.names.title ?? reference.lastPathComponent)
```

Comment:
I don't know if it's worth having a fallback value here. AFAIK the module node should always have a symbol so it may be better to add a comment and force unwrap here.

Reply:
I think I agree that force unwrapping is the right move here but maybe we should leave it as-is for now to reduce risk for 5.7? And just add a comment saying we think a force-unwrap should be appropriate here.